### PR TITLE
Merge 2.10.x into master.

### PR DESCRIPTION
**Pull request by paulp**
Notes on this merge:
- backported commits were avoided by using multiple merges and -Xours on backported ranges.
- b07228aebe is excluded from the merge because it breaks master, as recounted on internals.
- I don't know what is the significance or importance of the change to test/files/run/t6028.check.

Review by any willing party, @adriaanm, @retronym, etc. Don't make the merge commit stew - they don't improve with age.. Think of them as the bananas of the commit universe.

### CODE REVIEW

File: test/files/run/t6028.check [hunk 1]
```diff
@@ -15,7 +15,7 @@ package <empty> {
       }
     };
     def bar(barParam: Int): Object = {
-      @volatile var MethodLocalObject$module: runtime.VolatileObjectRef = new runtime.VolatileObjectRef(<empty>);
+      @volatile var MethodLocalObject$module: runtime.VolatileObjectRef = new runtime.VolatileObjectRef(null);
```
Comment (retronym) [inline test/files/run/t6028.check:18]: I believe this to be a progression due to #1956 /cc @JamesIry

**Comment by retronym** [comment]
lgtm, we've been busy little :bee:s

**Comment by paulp** [comment]
Oh, I'm so sick of this.

PLS REBUILD ALL

**Comment by paulp** [comment]
Manually verified at https://scala-webapps.epfl.ch/jenkins/job/scala-checkin-manual/766/console

File: test/files/run/t6028.check (hunk 1 above)
Comment (JamesIry) [inline test/files/run/t6028.check:18]: yup

**Pull request merged**